shared/src/main/java/com/couchbase/lite/Dictionary.java

```diff
@@ -97,6 +99,14 @@ public Dictionary set(Map<String, Object> dictionary) {
         return this;
     }
 
+    @Override
+    public List<String> getKeys() {
+        if (!changed)
+            return super.getKeys();
+        else
+            return allKeys();
```

Comment:
Both Dictionary.allKeys() and ReadOnlyDictonary.getKeys() create new instance of `List<String>`. It is not necessary.